free-3.md

```diff
@@ -10,41 +10,100 @@ _What need is met by your idea? Who are you helping? What is that person's
+are evaluated and the result is displayed in a visual form. The exact method of
+translating formulas to an image, I have yet to determine. I can think of
+several possibilities. The user could give formulas the determine the color as
+a function of space.
+
+Or, the formulas could be interpreted as describing a set of objects or
+surfaces in 3-space, which are then rendered to the 2D screen.
+
+Or, interesting properties of the functions specified the user could map to
+interesting features in the resulting image. For example, the zeros of a
+function could correspond to bright points or lines in the image.
+
+Or, these methods could be combined in a way specified by the user.
+
+In terms of errors, there are roughly two kinds. The first occurs when the user
+enters an ill-formed formula. This could be as simple as mismatched
+parentheses, or it could be more complex. The second type of error could occur
+when the formulas do not easily map to a visual result. For example,
+singularities may pose a problem.
+
+As this language is intended to be somewhat education, it would hopefully
+provide for all of these errors a helpful error message, which contains
+information about what went wrong, _why_ it went wrong, and how to fix it. It
+would be really awesome, although probably difficult, to link to a knowledge
+base like Wolfram Alpha where the user could get more information on the theory
+behind the error.
 
 ### Expressiveness
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+It should be easy to create a visually interesting picture which gives some insight about the formulas that created it. It should be difficult, but possible, to learn to predict the form of the image based on the formulas. But it should not be possible (or it should be prohibitively difficult) to choose formulas to create a specific image. Just as one would not use ContextFree to create a particular picture, so one should use this language instead for exploration.
 
 ### Related work
 _Are there any other DSLs in this domain? If not, describe how you know there
 aren't and conjecture why not. If so, describe them and provide links. How well 
 do they address the need? Are there any particularly admirable qualities of the
 language? Are there parts of the language you think could be improved?_
 
+I've mentioned that this language is like ContextFree in that it uses a formal
+theoretical concept to explore art, or vice versa. There are also many existing
+tools which can help the user visualize a mathematical function, but these are
+typically geared more towards academics. Examples include [MATLAB](http://www.m
+athworks.com/products/matlab/?requestedDomain=www.mathworks.com) and
+[Mathematica](https://www.wolfram.com/mathematica/). I do not know of any
+existing DSLs geared specifically towards the math-art boundary.
```

Comment:
The random art assignment from CS 131 Programming Languages is a lot like what you described here. It takes in mathematical expressions and outputs random art based on these expressions. They do it in Haskell, and all students do is define mathematical expressions to create interesting art.